addons/binding/org.openhab.binding.evohome/ESH-INF/thing/channels.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="evohome" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0" xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+	<channel-type id="systemMode">
+		<item-type>String</item-type>
+		<label>Mode</label>
+		<description>Current system mode</description>
+		<category>temperature</category>
+		<state pattern="%s">
+			<options>
+				<option value="Auto">Normal</option>
+				<option value="AutoWithEco">Eco</option>
+				<option value="Away">Away</option>
+				<option value="DayOff">Day off</option>
+				<option value="HeatingOff">Off</option>
+				<option value="Custom">Custom</option>
+			</options>
+		</state>
+	</channel-type>
+	<channel-type id="temperature">
+		<item-type>Number</item-type>
+		<label>Temperature</label>
+		<description>Current room temperature</description>
+		<category>temperature</category>
+		<state readOnly="true" pattern="%.1f °C">
+		</state>
+	</channel-type>
+	<channel-type id="currentsetpoint">
+		<item-type>Number</item-type>
+		<label>Current set point</label>
```

Comment:
Keep labels short and use capitalised strings. Would "Set Point" do here?

Reply:
I probably understand to little about how evohome works.
From openHAB perspective, any channel displays the "current" state by default, so this word should imho always be omitted. Furthermore, a "set point" - as the name suggest - is something to be "set"; what meaning does it have, when it is read-only? Shouldn't the set point and the override channel possibly be merged into one? I expect that if you override the set point, the "current set point" will show the value of your override, won't it?
Likewise you could get rid off the "cancel override" channel by e.g. defining that sending `NULL` to the setpoint cancels any override that might have been set.

Reply:
Let me explain a bit more, since I'm not too happy with the current interface as well and you might have a more elegant solution to offer. Set point in this context refers to the desired temperature of a particular room. This can be set from the evohome in any number of ways and is is represented here as "current setpoint". Then there are two ways of overriding this setpoint : permanently (currently implemented) and temporary (not implemented yet). Finally, you want to cancel that override.

You example works very well for the current implementation, however with the future temporary override in mind; this is the best alternative I could come up with. Do you see a more openHABesque alternative?

Reply:
How temporary is temporary? Is it time based (i.e. for 1 hour) or can certain events cancel it (e.g. if temperature falls below 10°C)?
If it is purely time based, my take would be that the temporary override does not make much sense to be exposed to openHAB at all - users would rather use openHAB to decide when to cancel a permanent override again (e.g. sending a cancel after one hour in a rule).

Reply:
It's time based. I can get on board with you reasoning. Let me see if I can set it up accordingly. Is sending `NULL` to a channel commonplace?